domain.go

```diff
@@ -26,6 +26,29 @@ type DomainInput struct {
 	Note        *string `json:"note,omitempty"`
 }
 
+func (d *Domain) GetTag(client *Client, tagId ID) (*Tag, error) {
+	if d == nil {
+		return nil, fmt.Errorf("Unable to GetTag() with nil Domain pointer")
+	}
+	tags, err := d.DomainId.Tags(client, nil)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting tags: %s", err)
+	}
+	tag, err := tags.GetTagById(tagId)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting tags: %s", err)
+	}
+	return tag, nil
+}
```

Comment:
This can be simplified as
```suggestion
	return tags.GetTagById(tagId)
}
```
 unless you thing the extra `Error getting tags` is helpful to the end user somehow?

Reply:
Wow.... this is a great call out 👏